src/Helpers/JWKFetcher.php

```diff
@@ -62,19 +68,44 @@ protected function convertCertToPem(string $cert) : string
         return $output;
     }
 
+    /**
+     * Get a specific kid from a JWKS.
+     *
+     * @param string      $kid     Key ID to get.
+     * @param string|null $jwksUri JWKS URI to use, or fallback on class-level one.
+     *
+     * @return mixed|null
+     */
+    public function getKey(string $kid, string $jwksUri = null)
+    {
+        $keys = $this->getKeys( $jwksUri );
+
+        if (! empty( $keys ) && empty( $keys[$kid] )) {
+            $keys = $this->getKeys( $jwksUri, false );
+        }
+
+        return $keys[$kid] ?? null;
+    }
+
     /**
      * Gets an array of keys from the JWKS as kid => x5c.
      *
-     * @param string $jwks_url Full URL to the JWKS.
+     * @param string  $jwks_url  Full URL to the JWKS.
+     * @param boolean $use_cache Set to false to skip cache check; default true to use caching.
      *
      * @return array
      */
-    public function getKeys(string $jwks_url) : array
+    public function getKeys(string $jwks_url = null, bool $use_cache = true) : array
     {
-        $cache_key = md5($jwks_url);
-        $keys      = $this->cache->get($cache_key);
-        if (is_array($keys) && ! empty($keys)) {
-            return $keys;
+        $jwks_url = $jwks_url ?? $this->guzzleOptions['base_uri'] ?? '';
```

Comment:
It turns out there is a `base_uri` setting in the HTTP library already so I'm leveraging that here. We need to know the final URI we're using here to check the cache (if necessary). I'm sticking with the "return empty array if URL is empty" because that is indicating "we don't have keys for that URL" and will fail with the "no kid found in that JWKS" error mentioned earlier. Same outcome for a malformed/empty JWKS.